cdap-spark-core-base/src/test/java/io/cdap/cdap/app/runtime/spark/serializer/KryoSerializerTest.java

```diff
@@ -65,10 +65,13 @@ public void testStructuredRecordSerializer() throws IOException {
       .set("double", 2.25d)
       .set("string", "Hello World")
       .set("bytes", "Hello Bytes".getBytes(StandardCharsets.UTF_8))
+      .set("ts", System.currentTimeMillis())
       .set("enum", "a")
       .set("array", new int[]{1, 2, 3})
       .set("map", ImmutableMap.of("1", 1, "2", 2, "3", 3))
-      .set("union", null).build();
+      .set("union", null)
+      .set("node", StructuredRecord.builder(schema.getField("node").getSchema()).build())
+      .build();
 
     Kryo kryo = new Kryo();
     kryo.addDefaultSerializer(Schema.class, SchemaSerializer.class);
```

Comment:
Would it make sense to add a unit test that checks the size of the serialized object to make sure we dont regress later on?

Reply:
It would be kind of flaky though (e.g. if kryo version upgrade, the size might be different).

Reply:
We could make it a bit fuzzy, like >10% change would cause a failure. But it might eventually become flaky anyway, so we can omit that for now